### PR TITLE
Changelog for Rust 1.67 :lady_beetle:

### PR DESCRIPTION
**Pull request by xFrednet**
Roses are red,
violets are blue,
if changelogs could talk,
what would we do?

---

The usual disclaimer: This PR is written, as if the version was already released. It should be merged with the coming release on 2023-01-26. So, please provide feedback and approve it, if everything looks good, but let's wait with the r+ until the release :)

---

changelog: none

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Comment by xFrednet** [comment]
I only fixed remark errors :)

**Comment by xFrednet** [comment]
R- since it is written like the changelog was already released. If we merge it now, it would say that the version was already released. Also, I might have missed the `beta-accepted` changes, now that I think of it :sweat_smile:

**Comment by xFrednet** [comment]
The changelog now also includes the backports :)

**Comment by Manishearth** [comment]
If you want to merge now you can change it to say 1.67 beta and we make a small edit once it's stable

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,11 +6,204 @@ document.
...
+* Moved [`or_fun_call`] to `nursery` (Now allow-by-default)
+  [#9829](https://github.com/rust-lang/rust-clippy/pull/9829)
+* Uplifted [`let_underscore_lock`] into rustc
+  [#9697](https://github.com/rust-lang/rust-clippy/pull/9697)
+* Uplifted [`let_underscore_drop`] into rustc
+  [#9697](https://github.com/rust-lang/rust-clippy/pull/9697)
+* Moved [`bool_to_int_with_if`] to `pedantic` (Now allow-by-default)
+  [#9830](https://github.com/rust-lang/rust-clippy/pull/9830)
+* [`manual_swap`]: No longer lints in const context
+  [#9871](https://github.com/rust-lang/rust-clippy/pull/9871)
```
Comment (giraffate) [inline CHANGELOG.md:57]: It would be better to move this to the `False Positive Fixes` section.

**Comment by xFrednet** [comment]
I changed the text, so we can merge it early and addressed the comment from @giraffate. IDK how the entry ended up there xD

**Review by giraffate — APPROVED** [review]
Thanks!

**Comment by flip1995** [comment]
I saw you've gone over the `beta-accepted` labels and included them. Thanks for this!

**Pull request merged**